zigzag/classes/hardware/architecture/AimcArrayUnit.py

```diff
@@ -0,0 +1,413 @@
+import numpy as np
+import math
+import copy
+if __name__ == "__main__":
+    from imc_unit import ImcUnit
+    from DimcArrayUnit import DimcArrayUnit
+    import logging as _logging
+    _logging_level = _logging.INFO
+    _logging_format = '%(asctime)s - %(funcName)s +%(lineno)s - %(levelname)s - %(message)s'
+    _logging.basicConfig(level=_logging_level,
+                         format=_logging_format)
+else:
+    import logging as _logging
+    from zigzag.classes.hardware.architecture.imc_unit import ImcUnit
+    from zigzag.classes.hardware.architecture.DimcArrayUnit import DimcArrayUnit
+
+###############################################################################################################
+# README
+#   . class AimcArrayUnit (defines the energy/area/delay cost of an ADC, a DAC and an AIMC array)
+# How to use this file?
+#   . This file is internally called in ZigZag-IMC framework.
+#   . It can also be run independently, for mainly debugging. An example is given at the end of the file.
+###############################################################################################################
+
+class AimcArrayUnit(ImcUnit):
```

Comment:
Why is it called "ArrayUnit"? Because it represents a "Macro"? If so I think it's more clear to rename it as such.